src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2604,7 +2605,23 @@ def aks_agentpool_delete(cmd, client, resource_group_name, cluster_name,
         raise CLIError("Node pool {} doesnt exist, "
                        "use 'aks nodepool list' to get current node pool list".format(nodepool_name))
 
-    return sdk_no_wait(no_wait, client.begin_delete, resource_group_name, cluster_name, nodepool_name)
+    active_cloud = get_active_cloud(cmd.cli_ctx)
+    if active_cloud.profile != "latest":
```

Comment:
Will this logic also be used in old profiles env such as Azure Stack?

Reply:
No, this feature is only used in AKS. @zhoxing-ms